.cirrus.yml

```diff
@@ -21,134 +24,158 @@ vm_instance_template: &VM_TEMPLATE
   region: eu-central-1
   subnet_id: ${CIRRUS_AWS_SUBNET}
 
-
-multi_arch_build_task: 
-  env:
-    matrix:
-      - version: 9/community
-        tag: $CURRENT_LTS_VERSION-community
-      - version: 9/developer
-        tag: $CURRENT_LTS_VERSION-developer
-      - version: 9/enterprise
-        tag: $CURRENT_LTS_VERSION-enterprise
-      - version: 9/datacenter/app
-        tag: $CURRENT_LTS_VERSION-datacenter-app
-      - version: 9/datacenter/search
-        tag: $CURRENT_LTS_VERSION-datacenter-search
-      - version: 10/community
-        tag: $NEXT_VERSION-community
-      - version: 10/developer
-        tag: $NEXT_VERSION-developer
-      - version: 10/enterprise
-        tag: $NEXT_VERSION-enterprise
-      - version: 10/datacenter/app
-        tag: $NEXT_VERSION-datacenter-app
-      - version: 10/datacenter/search
-        tag: $NEXT_VERSION-datacenter-search
+multi_arch_build_gcp_task: 
+  #only_if: $CIRRUS_CRON == 'nightly-mend-scan'
   ec2_instance:
     <<: *VM_TEMPLATE
-  login_script:
-    - docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
-  setup_script:
-    - docker run -t --rm --privileged tonistiigi/binfmt --install all
-    - docker buildx create --name multibuilder --driver docker-container
-    - docker buildx use multibuilder
-    - docker buildx inspect --bootstrap
-  build_script:
-    - echo "Build the ${STAGING_IMAGE_NAME}:${tag} image supporting two architectures, linux/amd64 and linux/arm64"
-    - .cirrus/multi-arch-build.sh ${STAGING_IMAGE_NAME} ${tag} ${version}
+  pull_public_images_script:
+    - echo "Pull the ${PUBLIC_IMAGE_NAME} datacenter public images"
+    - docker pull "${PUBLIC_IMAGE_NAME}:${CURRENT_VERSION}-datacenter-app"
+    - docker pull "${PUBLIC_IMAGE_NAME}:${CURRENT_VERSION}-datacenter-search"
+  login_to_gcr_script:
+    - export DOCKER_GCLOUD_PASSWORD=$(echo ${DOCKER_GCLOUD_SA_KEY} | base64 -d)
+    - docker login -u _json_key -p "$DOCKER_GCLOUD_PASSWORD" https://${GCLOUD_REGISTRY}
+  tag_and_promote_script:
+    - export CURRENT_MINOR_VERSION=$(echo ${CURRENT_VERSION} | cut -d '.' -f 1,2)
+    - echo "Promote the ${PUBLIC_IMAGE_NAME}:${CURRENT_VERSION}-datacenter-app images supporting two architectures (linux/amd64 and linux/arm64) to the GCP registry ${GCLOUD_REGISTRY}"
+    - docker tag "${PUBLIC_IMAGE_NAME}:${CURRENT_VERSION}-datacenter-app" "${GCLOUD_REGISTRY}/${GCLOUD_PRODUCT_NAME}/${PUBLIC_IMAGE_NAME}:${CURRENT_VERSION}-datacenter-app"
```

Comment:
here it should be:
docker tag "${PUBLIC_IMAGE_NAME}:${CURRENT_VERSION}-datacenter-app" "${GCLOUD_REGISTRY}/${GCLOUD_PRODUCT_NAME}:${CURRENT_VERSION}

As for the primary image (app) it should lay at the root of the repo, and tag should only be track numbers.

See [here](https://console.cloud.google.com/gcr/images/sonarqube-marketplace-provider/global/sonarqube-dce?project=sonarqube-marketplace-provider) and [here](https://console.cloud.google.com/gcr/images/cloud-marketplace/GLOBAL/agilebits-public/op-scim-bridge) as an example